### PR TITLE
AN-1074 (1075 & 1076) Initial push of silver txn tables

### PR DESCRIPTION
**Pull request by eric-laurello**

### CODE REVIEW

File: models/silver/silver__msgs.sql [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{{ config(
+  materialized = 'incremental',
+  unique_key = "CONCAT_WS('-', chain_id, block_id, tx_id, msg_index)",
+  incremental_strategy = 'delete+insert',
+  cluster_by = ['block_timestamp::DATE'],
```
Comment (desmond-hui) [inline models/silver/silver__msgs.sql:5]: This should be clustered by `ingested_at::date`

File: models/silver/silver__msgs.sql [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  tx_status,
+  INDEX AS msg_index,
+  VALUE :type :: STRING AS msg_type,
+  VALUE AS msg,
+  ingested_at
+FROM
+  {{ ref('silver__transactions') }} A,
+  LATERAL FLATTEN(
+    input => A.msgs -- ,
+    -- outer => TRUE
```
Comment (desmond-hui) [inline models/silver/silver__msgs.sql:23]: remove commented out code

File: models/silver/silver__msg_attributes.sql [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    blockchain,
+    chain_id,
+    tx_id,
+    msg_index,
+    msg_type,
+    SUBSTRING(path, POSITION('[', path) + 1, POSITION(']', path) - POSITION('[', path) -1) AS attribute_index,
+    key AS xKey,
+    TRY_BASE64_DECODE_STRING(
+      VALUE :: STRING
+    ) AS xValue,
```
Comment (desmond-hui) [inline models/silver/silver__msg_attributes.sql:22]: use something like `decoded_value` `decoded_key` is a bit more descriptive.  Also lets be consistent with casing.  We use all lower case with `_` between works.  So this should at least be `x_value` and `x_key`

File: models/silver/silver__msg_attributes.sql [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  p.chain_id,
+  p.tx_id,
+  p.msg_index,
+  p.msg_type,
+  p.attribute_index,
+  attribute_key,
+  attribute_value,
+  ingested_at
+FROM
+  base A pivot(MAX(xvalue) for xKey IN ('key', 'value')) AS p (
```
Comment (desmond-hui) [inline models/silver/silver__msg_attributes.sql:50]: what are you trying to achieve with `MAX(xvalue)`?  Do you have a sample tx_id where this is utilized?

**Review by desmond-hui — CHANGES_REQUESTED** [review]

File: models/silver/silver__msg_attributes.sql (hunk 4 above)
Comment (eric-laurello) [reply models/silver/silver__msg_attributes.sql:50]: I thought for the pivot you need some sort of an aggregate function. I'm just trying to get the two rows of data into two columns.

File: models/silver/silver__msg_attributes.sql (hunk 3 above)
Comment (eric-laurello) [reply models/silver/silver__msg_attributes.sql:22]: makes sense, will update.

**Comment by desmond-hui** [comment]
@eric-laurello are these failures expected?

`Failure in test silver__blocks__block_id-assert_no_gap (tests/silver/silver__blocks__block_id-assert_no_gap.sql)
23:43:45    Got 26 results, configured to fail if != 0
23:43:45  
23:43:45    compiled SQL at target/compiled/osmosis_models/tests/silver/silver__blocks__block_id-assert_no_gap.sql
23:43:45  
23:43:45  Failure in test silver__transactions__tx-gap (tests/silver/silver__transactions__tx-gap.sql)
23:43:45    Got 101 results, configured to fail if != 0
23:43:45  
23:43:45    compiled SQL at target/compiled/osmosis_models/tests/silver/silver__transactions__tx-gap.sql`

**Review by desmond-hui — APPROVED** [review]

**Pull request merged**